RFS/src/main/java/org/opensearch/migrations/bulkload/transformers/TransformerToIJsonTransformerAdapter.java

```diff
@@ -122,7 +132,8 @@ public GlobalMetadata transformGlobalMetadata(GlobalMetadata globalData) {
                 indexTemplates.stream()),
                 componentTemplates.stream()
                 )
-                .map(this::transformMigrationItem).collect(Collectors.toList());
+                .flatMap(item -> transformMigrationItem(item).stream())
+                .collect(Collectors.toList());
```

Comment:
There's some risk here. If a custom transform maps some components to OS2 and leaves others behind.  We'd really want/need to let version transforms run for some things and not others.  As discussed w/ Andre, adding a version key to each item would help us identify the version in the transform and also let subsequent transforms know that it's already been updated to that schema.